Extracting the price calculation logic allows us to test the
core logic of buyCoffee, but we still cannot test our charging
behaviour: that we are returning the correct values if a charge
is approved or declined!

diff --git a/04a-test_doubles/coffee-cart/src/cart.ts b/04a-test_doubles/coffee-cart/src/cart.ts
index c5e887a..0b6fafb 100644
--- a/04a-test_doubles/coffee-cart/src/cart.ts
+++ b/04a-test_doubles/coffee-cart/src/cart.ts
@@ -1,9 +1,14 @@
 import { BankCardReader } from "./bank";
 
-export function buyCoffee(price: number, modification?: string): string {
-	const reader = new BankCardReader();
-	let priceToCharge = price;
+export function getPrice(initialPrice: number, modification?: string): number {
+	let priceToCharge = initialPrice;
 	if (modification === "decaf") priceToCharge += 1.0;
 	else if (modification === "iced") priceToCharge += 0.5;
+	return priceToCharge;
+}
+
+export function buyCoffee(price: number, modification?: string): string {
+	const reader = new BankCardReader();
+	const priceToCharge = getPrice(price, modification);
 	return reader.charge(priceToCharge) ? "enjoy your coffee" : "card declined";
 }
diff --git a/04a-test_doubles/coffee-cart/test/cart.spec.ts b/04a-test_doubles/coffee-cart/test/cart.spec.ts
index 68647fb..52e6e0d 100644
--- a/04a-test_doubles/coffee-cart/test/cart.spec.ts
+++ b/04a-test_doubles/coffee-cart/test/cart.spec.ts
@@ -1,6 +1,19 @@
 import { expect } from "chai";
-import { buyCoffee } from "../src/cart";
+import { buyCoffee, getPrice } from "../src/cart";
 
+describe("getPrice", () => {
+	it("adds $1 for decaf", () => {
+		expect(getPrice(2, "decaf")).to.equal(3);
+	});
+	it("adds $0.5 for iced", () => {
+		expect(getPrice(2, "decaf")).to.equal(2.5);
+	});
+	it("returns same price with no mods", () => {
+		expect(getPrice(2)).to.equal(2);
+	});
+});
+
+// we still can't test the actual charging of the coffee though...
 describe("buyCoffee", () => {
 	it("serves coffee when the card is charged", () => {
 		expect(buyCoffee(4.5, "iced")).to.equal("enjoy your coffee");
